### PR TITLE
Fixes for includeChildMeshes:true when constructing physics shapes

### PR DESCRIPTION
**Pull request by eoineoineoin**
Allows MESH and CONVEX_HULL shapes to be constructed from TransformNode or InstancedMesh instances and  correctly account for transform of children when includeChildMeshes is set to true.

Discussed this with @carolhmj - there were a couple of failure cases or unexpected behaviour when building physics shapes.

Here's an example of one of the failure cases (check the console logs): https://playground.babylonjs.com/#5BJIYA#3 - this loads a glTF file which contains a mesh with multiple materials. The loader will create a single TransformNode with two Mesh children - one for each material.  If a user tries to build a physics shape from the "Cube" object (which is a glTF mesh, but a Babylon TransformNode) they'll hit the failure cases in the console.log()s. Similarly, if the mesh was instanced multiple times, the old code previously tried to use an `InstancedMesh` instance as if it were a `Mesh`, which would raise an exception accessing an undefined property (`Mesh.getVerticesData`)

This change allows a user to create a single physics shape from this scene construct. I've also tested this with some additional playground scenes, which @carolhmj linked me to:

https://playground.babylonjs.com/#5BJIYA#4
https://playground.babylonjs.com/#K7TJIG#363
https://playground.babylonjs.com/#9T64J6
https://playground.babylonjs.com/#R9DTZ0#16

And made a new variant of one, which tests this includeChildMeshes functionality in the complicated "Buggy" asset:
https://playground.babylonjs.com/#K7TJIG#371

Since I can imagine that this case of constructing a single physics shape from a whole branch of the scene hierarchy is useful, it would probably also be nice to expose the includeChildMeshes property to the PhysicsShapeMesh and PhysicsShapeConvexHull constructors, without having to require users use the PhysicsShape base class.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14030/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14030/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14030/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14030/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14030/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14030/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14030/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14030/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14030/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts [hunk 1]
```diff
@@ -52,18 +53,53 @@ class MeshAccumulator {
      * have a physics impostor. This is useful for creating a physics engine
      * that accurately reflects the mesh and its children.
      */
-    public addMesh(mesh: Mesh, includeChildren: boolean): void {
-        const indexOffset = this._vertices.length;
-        // Force absoluteScaling to be computed
+    public addNodeMeshes(mesh: TransformNode, includeChildren: boolean): void {
```
Comment (RaananW) [inline packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts:56]: This is probably a breaking change. You can keep the old function and proxy it to the new version while deprecating it (using the `@deprecated` tag in the tsdoc comment).

**Review by RaananW — COMMENTED** [review]
Just a small comment regarding back-compat. otherwise, LTGM! :-)

Run formatting (`npm run format:fix`) to fix the formatting/linting issue

File: packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts (hunk 1 above)
Comment (RaananW) [reply packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts:56]: Class is used internally, no breaking change.

**Review by sebavan — APPROVED** [review]

**Comment by RaananW** [comment]
@eoineoineoin - is this the full fix? or is there more to it?

**Pull request merged**